### PR TITLE
Remove remaining print()s from the library

### PR DESCRIPTION
**Pull request by booxter**
Use logger for any messages produced instead.

### CODE REVIEW

**Review by bcrochet — APPROVED** [review]
/lgtm

**Review by danmcp — APPROVED** [review]

**Review by danmcp — CHANGES_REQUESTED** [review]
Until #80 is fixed, we probably need to make this be:

```
if i == max retires-1:
  logger.error
  break
logger.debug
```

Otherwise we won't give any indication of an invalid setup until the very end.

Note: This suggestion is taking the rebase into account since the break wouldn't make sense as is.

File: src/instructlab/eval/mt_bench_common.py [hunk 1]
```diff
@@ -272,8 +272,7 @@ def chat_completion_openai(
             return response.choices[0].message.content
         except openai.OpenAIError as e:
             if i == API_MAX_RETRY - 1:
-                # Print error on last try
-                print(type(e), e)
+                logger.warning(e)
```
Comment (booxter) [inline src/instructlab/eval/mt_bench_common.py:275]: maybe even `error`? (@danmcp let me know if there's a good reason to do `info`, as in your snippet.)

Comment (booxter) [reply src/instructlab/eval/mt_bench_common.py:275]: I was! Will update to `error`.

**Review by nathan-weinberg — APPROVED** [review]
will rerun mypy check once https://github.com/instructlab/eval/pull/93 merges

**Comment by danmcp** [comment]
@booxter Note I rebased your commit to pull in #93

**Review by danmcp — APPROVED** [review]

**Pull request merged**